src/javascript/gameInstance.js

```diff
@@ -101,6 +102,32 @@ export default class GameInstance {
     }
   }
 
+  setUserKeyBind(trackKeyBind) {
+    const keyMap = this.vm.keyMap;
+    Logger.log("!!!", keyMap);
+    if (keyMap) {
+      this.userKeyBind = trackKeyBind.map((e) => keyMap[e]);
+      let ret = {};
+      Object.keys(keyMap).forEach((key) => {
+        ret[keyMap[key]] = key;
+      });
+      this.reverseKeyMap = ret;
+      Logger.log("keyMap:", this.userKeyBind, this.reverseKeyMap);
+    } else {
+      this.userKeyBind = trackKeyBind;
+      this.reverseKeyMap = null;
```

Comment:
Hmm I wonder if its better to set it as an empty object/map instead? Note: this change _might_ break behaviour if you rely on null checks for this instance variable...

```suggestion
      this.reverseKeyMap = {};
```

Reply:
Is this better in terms of type safety? I'll accept the change anyway, what could go wrong lul